brian2/parsing/bast.py

```diff
@@ -192,19 +204,53 @@ def render_Call(self, node):
         return node
 
     def render_BinOp(self, node):
+        node.left.parent = weakref.proxy(node)
+        node.right.parent = weakref.proxy(node)
         node.left = self.render_node(node.left)
         node.right = self.render_node(node.right)
         # TODO: we could capture some syntax errors here, e.g. bool+bool
         # captures, e.g. int+float->float
         newdtype = dtype_hierarchy[max(dtype_hierarchy[subnode.dtype] for subnode in [node.left, node.right])]
+        if node.op.__class__.__name__ == 'Div':
+            # Division turns integers into floating point values
+            newdtype = 'float'
+            # Give a warning if the code uses floating point division where it
+            # previously might have used floor division
+            if node.left.dtype == node.right.dtype == 'integer':
```

Comment:
Is it worth also warning on things like ``True/2`` that would have returned 0 but now 0.5? In other words, dealing with bool as well as integer? Actually, we might like to think about that throughout?

Reply:
I think doing arithmetic operations with integers directly should raise an error, that's why we introduced the `int` function in the first place (that was back in the days when Achilleas worked on BrianDroid, Java does not support this kind of arithmetic). Let's deal with this in a separate PR?

Reply:
With bools you mean? OK.

Reply:
Sorry, yes, with bools.